docs_app/content/guide/testing/marble-testing.md

```diff
@@ -202,7 +203,46 @@ it('should repeat forever', () => {
 });
 ```
 
-***
+## Synchronous Assertion
+
+Sometimes, we need to assert changes in state after an observable stream has completed - such as when our subscription doesn’t map to another observable but instead just updates a basic variable. Outside of Marbles testing with TestScheduler, we might think of this as creating a delay or setting a timer before making our assertion.
+
+For example:
+
+```ts
+it('should set a local variable', () => {
+  let myVar = 0;
+
+  const testScheduler = new TestScheduler(() => {});
+
+  testScheduler.run(helpers => {
+    const { cold, flush } = helpers;
+
+    const myStream$ = cold('--a--b|', {
+      a: 'value emitted',
+      b: 'another value emitter',
+    });
+
+    myStream$
+      .pipe(
+        map(() => {
+          myVar++;
+        }),
+      )
+      .subscribe();
+
+    // trigger testScheduler 'virtual time' to complete all outstanding hot or cold observables
+    // (or, 'Pause' in virtual time until the observable completes)
+    flush();
+
+    expect(myVar).toBe(2);
+  });
+});
+```
```

Comment:
If an example of a test that explicitly calls `flush` is to be included, it will have to, IMO, be a better test than this. This test doesn't test anything that could not be better tested with `expectObservable`.

Reply:
Hi @cartant 
Thanks for reviewing my PR so quickly.

1. I had previously tried to keep my example test as short/simple as possible, but now agree with you that it needs to be clearer. I've submitted an update for your review.
2. Can I take it that you're ok with my suggested text changes, and all I need to do is iterate on an example test to get your approval?

Thanks
Tom

Reply:
I've only skimmed the text changes. For me, the example test is more important, as it's, IMO, important that it's idiomatic and encourages good testing practice.

I'm still not convinced that the example is realistic or that it represents testing best practice. I would not want to encourage people to test using state outside of the `it` call.

I think a better example would be something like what's done in the library's tests. Something like this:

https://github.com/ReactiveX/rxjs/blob/01a09789a0a9484c368b7bd6ed37f94d25490a00/spec/operators/finalize-spec.ts#L79-L88

Reply:
@cartant ALL of our testing documentation should really be guiding people to use "run mode" tests, what we're doing in the library here really needs to be refactored to do the same as soon as possible.

So it should look like:

```ts
const rxTest = new TestScheduler(assertDeepEquals);
rxTest.run(({ hot, cold, expectObservable }) => {
  /// tests here.
});
```

Reply:
Ah, nevermind, I see what you were saying. You weren't suggesting globals.